resources/assets/lib/stores/channel-store.ts

```diff
@@ -200,9 +218,15 @@ export default class ChannelStore implements DispatchListener {
   private handleChatMessageNewEvent(event: MessageNewEvent) {
     for (const message of event.json.messages) {
       const channel = this.channels.get(message.channel_id);
-      if (channel == null) continue;
 
-      channel.addMessage(message);
+      if (channel != null) {
+        channel.addMessage(message);
+      } else if (!this.ignoredChannels.has(message.channel_id)) {
+        getChannel(message.channel_id)
+          .done((json) => this.update(json))
+          // FIXME: this seems like the wrong place to tigger an error popup.
```

Comment:
```suggestion
          // FIXME: this seems like the wrong place to trigger an error popup.
```

also probably should ignore 404/403 errors for when the user isn't in the channel anymore